tests/integration_test.rs

```diff
@@ -94,6 +95,72 @@ fn instrument_dfinity_with_fault_injection() {
         };
     }
 }
+
+#[test]
+fn instrument_handwritten_wasm() {
+    // executable is located at target/debug/whamm
+    let executable = "target/debug/whamm";
+
+    // if you want to change the wat file
+    // (calling wat2wasm from a child process doesn't work
+    //  since somehow the executable can't write to the file system directly)
+    let file_data = fs::read("tests/apps/handwritten/add.wat").unwrap();
+    let wasm_data = wat2wasm(file_data).unwrap();
+    fs::write("tests/apps/handwritten/add.wasm", wasm_data).unwrap();
+
+    let res = Command::new(executable)
+        .arg("instr")
+        .arg("--script")
+        .arg("tests/scripts/instr.mm")
+        .arg("--app")
+        .arg("tests/apps/handwritten/add.wasm")
+        .output()
+        .expect("failed to execute process");
+    assert!(res.status.success());
+
+    let out = Command::new("wasm2wat")
+        .arg("output/output.wasm")
+        .output()
+        .expect("failed to execute process");
+
+    let stdout = String::from_utf8(out.stdout).unwrap();
+    println!("{}", stdout);
+}
+
+#[test]
+fn instrument_control_flow() {
+    let executable = "target/debug/whamm";
+
+    // run cargo run on control flow
+    let a = Command::new("cargo")
+        .arg("build")
+        .arg("--target")
+        .arg("wasm32-unknown-unknown")
+        .current_dir("wasm_playground/control_flow")
+        .output()
+        .expect("failed to execute process");
+    assert!(a.status.success());
+
+    let res = Command::new(executable)
+        .arg("instr")
+        .arg("--script")
+        .arg("tests/scripts/instr.mm")
+        .arg("--app")
+        .arg("wasm_playground/control_flow/target/wasm32-unknown-unknown/debug/cf.wasm")
+        .output()
+        .expect("failed to execute process");
+    assert!(res.status.success());
+
+    let out = Command::new("wasm2wat")
+        .arg("output/output.wasm")
+        .output()
+        .expect("failed to execute process");
+
+    // write to file
+    let stdout = String::from_utf8(out.stdout).unwrap();
+    fs::write("output/output.wat", stdout).unwrap();
```

Comment:
You wouldn't need to write this to a file, we've already done assertions at this point. All we care about is the `expect` above. Also, if you have `wabt` included as a dependency, is there a `wasm2wat` utility we can use instead? Like you used `wat2wasm` above?

Reply:
I was writing to file for easier manual inspection of the instrumentation behavior.